FrejaEidClient/src/main/java/com/verisec/frejaeid/client/service/AuthenticationService.java

```diff
@@ -101,4 +119,30 @@ private boolean isPollingTimeExpired(long pollingEndTime) {
         return (System.currentTimeMillis() + pollingTimeoutInMilliseconds) < pollingEndTime;
     }
 
+    public static byte[] readAllBytes(InputStream inputStream) throws IOException {
+        final int bufLen = 4 * 0x400; // 4KB
+        byte[] buf = new byte[bufLen];
+        int readLen;
+        IOException exception = null;
+
+        try {
+            try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+                while ((readLen = inputStream.read(buf, 0, bufLen)) != -1)
+                    outputStream.write(buf, 0, readLen);
+
+                return outputStream.toByteArray();
+            }
+        } catch (IOException e) {
+            exception = e;
+            throw e;
+        } finally {
+            if (exception == null) inputStream.close();
+            else try {
+                inputStream.close();
```

Comment:
This looks like a side-effect of this method. I wouldn't expect that the method closes input stream that was passed as argument, since the name of the method is readAllBytes, and also it wasn't initially opened by this method.

Reply:
Agreed, that part of the code was removed and the entire function simplified a bit. Done.